### PR TITLE
 Explicitly select global interfaces version

### PR DESCRIPTION
**Pull request by novakne**
Currently works on all interfaces instead of just global

### CODE REVIEW

File: build.zig [hunk 1]
```diff
@@ -94,6 +97,17 @@ pub const ScanProtocolsStep = struct {
         ) catch unreachable);
     }
 
+    pub fn interfaces(self: *ScanProtocolsStep, ifaces: []const scanner.RequiredInterface) void {
+        for (ifaces) |iface| {
+            // wl_display, wl_registry, and wl_callback are always generated
+            // as their version cannot change.
+            if (mem.eql(u8, iface.name, "wl_display") or mem.eql(u8, iface.name, "wl_registry") or
+                mem.eql(u8, iface.name, "wl_callback")) continue;
```
Comment (ifreund) [inline build.zig:105]: We should print an error message here and panic instead of ignoring these, also this logic belongs in `scanner.zig` IMO.

**Review by ifreund — COMMENTED** [review]
I don't want the user to need to specify all interfaces, only globals. To achieve this we could do something like the following:

1. create an initial list of required interfaces including only the globals.
2. start generating code, poping interfaces from the list as they are generated.
3. for every interface that is referenced (e.g. as the argument of a request) in the generated code push that interface to the list of required interfaces.
3. Continue generating code, poping generated interfaces and pushing referenced interfaces to the required list until the list of referenced interfaces is empty.

How's that sound?

**Comment by novakne** [comment]
So it does works as expected but this is kinda ugly and currently the client could just doesn't add `scanner.interfaces()` in its `build.zig` and it would works anyway so not ideal

**Comment by novakne** [comment]
Just to sum up where this PR is:

Using `scanner.interfaces()` will generate bindings only for the requested interfaces, e.g.
```zig
// build.zig
scanner.interfaces(&.{
    .{ .name = "wl_output", .version = 3 },
});
```
will not generate the events `name` and `description`, that part works as we want.

The problem is the `scanner.interfaces()` is not required, so doesn't add much safety.

Getting a list of required globals interfaces is hard, using arguments of a request would add a lot of interfaces, not only globals.

Other things is once we get a way to know globals interfaces we would need to add it to the lists of required_interfaces, that is as a struct `RequestedInterface{. {.name = "foo", .version = 1} }`, because I can't see how would we compare it to the client list otherwise. 
So what would we use? A `TailQueue()`, wich mean we should create a node everytime, doesn't sounds too ideal to me.

**Comment by ifreund** [comment]
Thanks for starting the work on this, I've ended up going a slightly different direction with this: #35. Feedback welcome as always!

**Pull request closed without merging**